kerlescan/view_helpers.py

```diff
@@ -54,86 +54,91 @@ def ensure_account_number(request, logger):
         raise HTTPError(HTTPStatus.BAD_REQUEST, message="identity not found on request")
 
 
+def check_request_from_drift_service(**kwargs):
+    """
+    check_request_from_drift_service kwargs need to contain: request
+    This method check if the request comes from our drift service.
+    """
+    request = kwargs["request"]
+    auth_key = get_key_from_headers(request.headers)
+
+    auth = json.loads(base64.b64decode(auth_key))
+    identity_type = auth.get("identity", {}).get("type", None)
+    if identity_type in ["System", "Associate"]:
```

Comment:
This is going to fail for our case. `System` will have `x-rh-internal-api` header so it will be able to check for shared secret but `Associate` will have it only if we explicitly do it with request between our own services. The first request coming from "outside" through Turnpike is going to be `Associate` but it will not have `x-rh-internal-api` header at all.

I recommend to split the checks into two branches by `identity_type` and one will check the `x-rh-internal-api` and the other will not. Also that would mean the naming of `check_request_from_drift_service` would be unfortunate. You can also remove the `Associate` part from here and have another method, perhaps named `check_request_from_turnpike`.